core/runtime/src/main/java/io/quarkus/runtime/configuration/ApplicationPropertiesConfigSource.java

```diff
@@ -87,4 +89,44 @@ private static InputStream openStream() {
             }
         }
     }
+
+    public static final class LocalConfigFile extends ApplicationPropertiesConfigSource {
+        private static final Logger LOGGER = Logger.getLogger(LocalConfigFile.class.getName());
+
+        public LocalConfigFile() {
+            super(openStream(), 290);
+        }
+
+        private static InputStream openStream() {
+            final Optional<String> localConfigFile = getLocalConfigFilename();
+            if (localConfigFile.isPresent()) {
+
+                final Path localConfigFilePath = Paths.get(localConfigFile.get());
+
+                if (Files.notExists(localConfigFilePath)) {
+                    LOGGER.warnf("Supplied local configuration file %s does not exist.", localConfigFilePath);
+                    return null;
+                }
+
+                try {
+                    return Files.newInputStream(localConfigFilePath);
+                } catch (NoSuchFileException | FileNotFoundException e) {
+                    return null;
+                } catch (IOException e) {
+                    throw new IOError(e);
+                }
+            }
+
+            return null;
+        }
+
+        public static Optional<String> getLocalConfigFilename() {
```

Comment:
Shouldn't there be a default to `${user.home}/.application.properties`?  @lbernardomaia did I miss the purpose of the change?

Reply:
Hey @dmlloyd I missed updating this part. After giving it a bit of thoughts maybe we should drop the default value and just use the value that will be defined by the user by using the `-Dquakurs.local-config` property?

Reply:
I guess then the config source would not be included if there was no name defined by the property?

Reply:
Yes, it will be just like the FileSystem app config source which returns an empty Properties object.

Reply:
OK.  I'm just not 100% sure that we're really solving a real problem, or at least the real problem that the user originally encountered.  It sounds like what we're really doing here is the same as specifying a path for the existing filesystem-backed config source.

Reply:
> OK. I'm just not 100% sure that we're really solving a real problem, or at least the real problem that the user originally encountered. 

I'd be interested to hear what @lbernardomaia thinks about this and if it covers the use case described in the PR.